libdnf/dnf-repo.c

```diff
@@ -746,6 +772,33 @@ dnf_repo_set_keyfile_data(DnfRepo *repo, GError **error)
 
     g_debug("setting keyfile data for %s", priv->id);
 
+    /* enabled is optional */
+    if (g_key_file_has_key(priv->keyfile, priv->id, "enabled", NULL)) {
+        if (dnf_repo_get_boolean(priv->keyfile, priv->id, "enabled", NULL))
+            enabled |= DNF_REPO_ENABLED_PACKAGES;
+    } else {
+        enabled |= DNF_REPO_ENABLED_PACKAGES;
+    }
+
+    /* enabled_metadata is optional */
+    if (g_key_file_has_key(priv->keyfile, priv->id, "enabled_metadata", NULL)) {
+        if (dnf_repo_get_boolean(priv->keyfile, priv->id, "enabled_metadata", NULL))
+            enabled |= DNF_REPO_ENABLED_METADATA;
+    } else {
+        g_autofree gchar *basename = g_path_get_basename(priv->filename);
+        /* special case the satellite and subscription manager repo */
+        if (g_strcmp0(basename, "redhat.repo") == 0)
```

Comment:
This should be part of a libdnf plugin instead of in libdnf itself, shouldn't it?